smart-contracts/test/testBridgeBank.js

```diff
@@ -169,4 +170,241 @@ describe("Test Bridge Bank", function () {
+      // expect token1 to be registered as a BridgeToken
+      isInCosmosWhitelist1 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token1.address);
+      expect(isInCosmosWhitelist1).to.be.true;
+
+      // expect token2 to be registered as a BridgeToken
+      isInCosmosWhitelist2 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token2.address);
+      expect(isInCosmosWhitelist2).to.be.true;
+
+      // expect token3 to be registered as a BridgeToken
+      isInCosmosWhitelist3 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token3.address);
+      expect(isInCosmosWhitelist3).to.be.true;
+    });
+
+    it("should NOT allow a user to add many BridgeTokens in a batch", async function () {
+      // expect token1 to NOT be registered as a BridgeToken
+      let isInCosmosWhitelist1 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token1.address);
+      expect(isInCosmosWhitelist1).to.be.false;
+
+      // expect token2 to NOT be registered as a BridgeToken
+      let isInCosmosWhitelist2 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token2.address);
+      expect(isInCosmosWhitelist2).to.be.false;
+
+      // expect token3 to NOT be registered as a BridgeToken
+      let isInCosmosWhitelist3 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token3.address);
+      expect(isInCosmosWhitelist3).to.be.false;
+
+      // add tokens as BridgeTokens
+      await expect(state.bridgeBank.connect(userOne)
+        .batchAddExistingBridgeTokens([
+          state.token1.address,
+          state.token2.address,
+          state.token3.address
+        ])
+      ).to.be.revertedWith('!owner');
+
+      // check if the tokens are now registered (should not be)
+      // expect token1 to NOT be registered as a BridgeToken
+      isInCosmosWhitelist1 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token1.address);
+      expect(isInCosmosWhitelist1).to.be.false;
+
+      // expect token2 to NOT be registered as a BridgeToken
+      isInCosmosWhitelist2 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token2.address);
+      expect(isInCosmosWhitelist2).to.be.false;
+
+      // expect token3 to NOT be registered as a BridgeToken
+      isInCosmosWhitelist3 = await state.bridgeBank.getCosmosTokenInWhiteList(state.token3.address);
+      expect(isInCosmosWhitelist3).to.be.false;
+    });
+
+    it("should allow anyone to forceSetBridgeTokenDenom", async function () {
+      // expect token2's denom to NOT be registered in BridgeBank
+      let denomInBridgeBank = await state.bridgeBank.contractDenom(state.token2.address);
+      expect(denomInBridgeBank).to.be.equal("");
+
+      // add token 2 as BridgeToken
+      await state.bridgeBank.connect(owner).addExistingBridgeToken(state.token2.address);
+
+      // userOne calls forceSetBridgeTokenDenom
+      await expect(state.bridgeBank.connect(userOne)
+        .forceSetBridgeTokenDenom(state.token2.address))
```

Comment:
Let's add a test where we try to call `forceSetBridgeTokenDenom` with a non cosmos whitelisted asset, and assert that this fails.